### PR TITLE
[ISSUE-677]: Proposal for CSI volume health feature support

### LINKED ISSUE #677: Proposal for CSI volume health feature support
**Is your feature request related to a problem? Please describe.**
Need to come up with the proposal for CSI volume Health feature support - https://kubernetes.io/docs/concepts/storage/volume-health-monitoring/

**Describe the solution you'd like**
Need to provide design notes about supporting volume health feature using k8s volume health monitor mechanism

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Aber4Nod**
### Resolves #677 

Providing design notes about supporting volume health feature using k8s volume health monitor mechanism

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Provide test details_

### CODE REVIEW

File: docs/proposals/volume-health-monitoring.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+Last updated: 24.12.2021
+
+## Abstract
+
+This proposal contains approaches for supporting CSI volume health feature, using Kubernetes volume health monitor mechanism. 
+
+## Background
+
+Currently, Baremetal CSI does not support any general conventional way to monitor Persistent Volumes after they are provisioned in Kubernetes.
```
Comment (mishoyama) [inline docs/proposals/volume-health-monitoring.md:11]: It does. Health and Missing operational status of PVs are exposed through the Volumes CRD

File: docs/proposals/volume-health-monitoring.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+This makes it very hard to debug and detect root causes when something happens to the volumes. 
+An application may detect that it can no longer write to volumes. However, the root cause happened at the underlying storage 
+system level. This requires several teams jointly debug and find out what has triggered the problem. 
+It could be that the volume runs out of capacity and needs an expansion.
```
Comment (mishoyama) [inline docs/proposals/volume-health-monitoring.md:15]: This is not a `health` state. Some volumes are always out of space - it depends on application logic.

File: docs/proposals/volume-health-monitoring.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+This proposal contains approaches for supporting CSI volume health feature, using Kubernetes volume health monitor mechanism. 
+
+## Background
+
+Currently, Baremetal CSI does not support any general conventional way to monitor Persistent Volumes after they are provisioned in Kubernetes.
+This makes it very hard to debug and detect root causes when something happens to the volumes. 
+An application may detect that it can no longer write to volumes. However, the root cause happened at the underlying storage 
+system level. This requires several teams jointly debug and find out what has triggered the problem. 
+It could be that the volume runs out of capacity and needs an expansion. 
+It could be that the volume was deleted by accident outside of Kubernetes.
```
Comment (mishoyama) [inline docs/proposals/volume-health-monitoring.md:16]: This case is partially covered by missing operational state.

File: docs/proposals/volume-health-monitoring.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+## Background
+
+Currently, Baremetal CSI does not support any general conventional way to monitor Persistent Volumes after they are provisioned in Kubernetes.
+This makes it very hard to debug and detect root causes when something happens to the volumes. 
+An application may detect that it can no longer write to volumes. However, the root cause happened at the underlying storage 
+system level. This requires several teams jointly debug and find out what has triggered the problem. 
+It could be that the volume runs out of capacity and needs an expansion. 
+It could be that the volume was deleted by accident outside of Kubernetes. 
+In the case of local storage, the local PVs may not be accessed any more due to the nodes break down.
```
Comment (mishoyama) [inline docs/proposals/volume-health-monitoring.md:17]: When node is down scheduler will exclude it from the list. We also have node health monitor in place which deletes ACs on down nodes.

File: docs/proposals/volume-health-monitoring.md (hunk 1 above)
Comment (Aber4Nod) [reply docs/proposals/volume-health-monitoring.md:11]: Yes, and a bit later this was described, but it is not the standardized approach, which is native to k8s mechanisms (e.g. events, pvc, pods). The client/customer should know about CSI's own CRD (e.g. it's how implemented DR SP on operator)

File: docs/proposals/volume-health-monitoring.md (hunk 2 above)
Comment (Aber4Nod) [reply docs/proposals/volume-health-monitoring.md:15]: This sentence is about possible cases, and there is no mapping to health about this sentence

File: docs/proposals/volume-health-monitoring.md (hunk 4 above)
Comment (Aber4Nod) [reply docs/proposals/volume-health-monitoring.md:17]: Yes, it is. But it is not related to monitoring

File: docs/proposals/volume-health-monitoring.md (hunk 3 above)
Comment (Aber4Nod) [reply docs/proposals/volume-health-monitoring.md:16]: And again, it's about some logic over it, not standartized way to handle it with k8s native resources

**Review by yurkov-anton — APPROVED** [review]

**Review by rkiselenko — APPROVED** [review]

**Pull request merged**